buildroot/share/PlatformIO/scripts/common-dependencies.py

```diff
@@ -212,9 +173,19 @@ def apply_features_config():
 				blab("Running extra_scripts for %s... " % feature, 2)
 				env.SConscript(feat['extra_scripts'], exports="env")
 
-			if 'custom_marlin_src_filter' in feat:
-				blab("========== Adding custom_marlin_src_filter for %s... " % feature, 2)
-				append_src_filter(feat['custom_marlin_src_filter'])
+			if 'src_filter' in feat:
+				blab("========== Adding build_src_filter for %s... " % feature, 2)
+				src_filter = ' '.join(env.GetProjectOption('build_src_filter'))
+				# first we need to remove the references to the same folder
+				my_srcs = re.findall(r'[+-](<.*?>)', feat['src_filter'])
+				cur_srcs = re.findall(r'[+-](<.*?>)', src_filter)
+				for d in my_srcs:
+					if d in cur_srcs:
+						src_filter = re.sub(r'[+-]' + d, '', src_filter)
+
+				src_filter = feat['src_filter'] + ' ' + src_filter
+				set_env_field('build_src_filter', [src_filter])
+				env.Replace(BUILD_SRC_FILTER=src_filter)
```

Comment:
Inside the building environment (SCons), we still use `SRC_FILTER` to keep compatibility with the existing projects that use advanced scripting.